modules/internal/ChapelArray.chpl

```diff
@@ -3211,6 +3211,68 @@ module ChapelArray {
       else
         a.add(e);
   }
+
+  //
+  // BaseSparseDom operator overloads
+  //
+  proc +=(ref sd: domain, inds: [] sd.idxType) where isSparseDom(sd) &&
+    sd.rank==1 {
+
+    if inds.size == 0 then return;
+
+    sd._value.dsiBulkAdd(inds);
+  }
+
+  proc +=(ref sd: domain, inds: [] sd.rank*sd.idxType) where isSparseDom(sd) &&
+    sd.rank>1 {
+
+    if inds.size == 0 then return;
+
+    sd._value.dsiBulkAdd(inds);
+  }
```

Comment:
Can these two overloads be combined into one by replacing the element type on the `inds` arguments with `index(sd)`  ("the index type of `sd`")?

I may be missing something, but it seems like these new `+=` operators were added out of thin air in this PR without explanation (the PR message suggests to me that they are pre-existing, but I can't see any indication of that — has there been support for them in the past that I'm overlooking?).  Was there any design discussion on adding support for them?  Should they be documented?